src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs

```diff
@@ -203,7 +241,35 @@ private UIElementCollection GetDefs()
 
 		private static Rect GetPathBoundingBox(Shape shape)
 		{
-			return shape._mainSvgElement.GetBBox();
+			if (FeatureConfiguration.Shape.CacheBBoxCalculationResult)
+			{
+				var key = shape switch
+				{
+					// most other properties don't impact getBBox result, see also comments in UpdateRender.
+					Path path => $"path:{(path.Data as GeometryData)?.Data}",
+					Line line => $"line:{line.X1},{line.Y1}-{line.X2}{line.Y2}",
+					Polygon polygon => $"polygon:{polygon.Points?.ToCssString()}",
+					Polyline polyline => $"polyline:{polyline.Points?.ToCssString()}",
```

Comment:
I wonder, should we cache this further? Maybe by computing those strings ahead of time?